### PR TITLE
Add BSGS for new field PVSS

### PR DESCRIPTION
**Pull request by waamm**
## Description
- Removes `ThresholdConfig` trait from `WeightedConfig`
- Moves most of `aptos-dkg`'s `arkworks` code from using `ark_std::rand` for randomness, to our (older) crate `rand`
- Removes Fiat-Shamir entirely from `das` (weighted and unweighted), now it uses `thread_rng()` to generate random numbers.
- Simplified the remaining FS code since we now no longer use it for `blstrs::Scalar`
- Added a `generate()` method to existing range proof and sigma proof code, in order to make the `generate()` method of the new PVSS cleaner.
- Removed the `sample()` method from GroupGenerators and replaces it with `default()` since we can't sample securely at the moment, and don't really seem to need it anyway.
- Adds the baby-step giant-step algorithm.
- Extended sigma protocol to lifted homomorphisms
- Added a `change_lifetime` method to sigma proofs, so it can be made independent of the homomorphism
- Add `pp: &Self::PublicParameters` to `decrypt_own_share()`, this reverberates through a lot of code

## How Has This Been Tested?
- Existing `aptos-dkg` tests
- A new test for the baby-step giant-step algorithm.

## Key Areas to Review
- It's important that the new random number generation for `das` is secure.
- `aptos-core/types` had to be modified due to the `decrypt_own_share()` change, hopefully not an issue.

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [x] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [x] Tests

## Which Components or Systems Does This Change Impact?
- [x] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Move Compiler
- [x] Other: most of this code is not yet in use

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [x] I tested both happy and unhappy path of the functionality
- [x] I have made corresponding changes to the documentation

---

> [!NOTE]
> Introduces a BSGS dlog helper, switches DKG/arkworks code to rand-based RNGs with new random utilities, removes Fiat‑Shamir from DAS, adds pp to decrypt_own_share, and extends sigma/range proof infra (incl. generators), updating benches/tests/types accordingly.
> 
> - **Core crypto/utils**:
>   - Add `dlog/bsgs` (baby-step giant-step) with table builder and tests.
>   - New `aptos_crypto::arkworks::random` utilities (UniformRand, sample_field_element[s], unsafe_random_point[s]).
>   - Replace `ark_std::rand` with `rand` across modules; `GroupGenerators` now `Default()` (generator-based) instead of `sample()`.
> - **PVSS (DAS and traits)**:
>   - Remove Fiat–Shamir from DAS verification; use `thread_rng()`/random scalars.
>   - Change `Transcript::decrypt_own_share(...)` signature to include `pp: &Self::PublicParameters`; propagate through weighted/unweighted PVSS, generic weighting, tests, benches, and `types/real_dkg`.
>   - Delete obsolete `pvss/das/fiat_shamir.rs`.
> - **Sigma protocol & homomorphisms**:
>   - Support lifted homomorphisms in fixed-base MSMs; implement `Trait` for `LiftHomomorphism` and add proper DST composition.
>   - Add `Proof::change_lifetime` and enable verification with generic hom types.
>   - Use `rand_core` bounds consistently in witnesses and proofs.
> - **Range proofs / KZG**:
>   - Update hiding KZG: redefine `Commitment` and `CommitmentRandomness` (alias to `Scalar`), adapt open/verify.
>   - Extend Univariate DeKART (v1, v2): switch RNG, add proof `generate()` helpers, minor API tweaks (commit randomness handling), FS helpers streamlined.
> - **Benches/Tests**:
>   - Update benches to `GroupGenerators::default()` and new APIs; add random proof generators.
>   - Adjust tests for new `decrypt_own_share` param and RNG changes; add BSGS test.
> - **Misc**:
>   - Remove `ThresholdConfig` impl from `WeightedConfig`.
>   - Dependency bumps (e.g., `fastrand` 2.3.0, `proc-macro2` 1.0.103).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 013d7f178b7a39878179d95403a209cfb18aaa3e. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: crates/aptos-dkg/src/pvss/das/weighted_protocol.rs [hunk 1]
```diff
@@ -199,17 +199,12 @@ impl traits::Transcript for Transcript {
...
-            eks,
-            auxs,
-            &Self::dst(),
-            2 + W * 3, // 3W+1 for encryption check, 1 for SoK verification.
+        let mut rng = rand::thread_rng();
+        let f = random_scalars(
+            sc.get_total_num_shares() + 1 - sc.tc.get_threshold(),
+            &mut rng,
         );
+        let extra = random_scalars(2 + W * 3, &mut rng);
```
[the remark this answers is not in the document]
Comment (waamm) [reply crates/aptos-dkg/src/pvss/das/weighted_protocol.rs:203]: > allowing malicious provers to potentially find challenges that make invalid transcripts pass verification

No, the challenges are derived by the verifier.

File: crates/aptos-crypto/src/arkworks/random.rs [hunk 2]
```diff
@@ -7,9 +7,36 @@
...
+    R: rand_core::RngCore + rand_core::CryptoRng,
+{
+    let r: C::ScalarField = sample_field_element(rng);
+
+    C::generator().mul(r)
+}
+
+/// Samples `n` uniformly random elements from the group, but is unsafe in the sense
+/// that the caller learns the discrete log of the random point.
+pub fn unsafe_random_points<C: CurveGroup, R>(n: usize, rng: &mut R) -> Vec<C>
```
Comment (alinush) [inline crates/aptos-crypto/src/arkworks/random.rs:33]: Where do you need this and the one above btw?

File: crates/aptos-dkg/src/dlog/bsgs.rs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+/// Compute discrete log using baby-step giant-step with a precomputed table
+///
+/// # Arguments
+/// - `G`: base of the exponentiation
+/// - `H`: target point
+/// - `baby_table`: precomputed HashMap from `C::Affine.to_compressed()` |---> exponent
+/// - `m`: size of the baby-step table
+#[allow(non_snake_case)]
+pub fn baby_step_giant_step<C: CurveGroup, const N: usize>(
```
Comment (alinush) [inline crates/aptos-dkg/src/dlog/bsgs.rs:15]: Wonder whether `N` could be inferred from `CurveGroup` here.

File: crates/aptos-dkg/src/dlog/bsgs.rs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    None
+}
+
+/// Build a baby-step table of size `m`
+///
+/// Returns a HashMap: `C::Affine.to_compressed() |---> exponent`
+#[allow(non_snake_case)]
+pub fn build_baby_table<C: CurveGroup, const N: usize>(G: C, m: u64) -> HashMap<[u8; N], u64> {
```
Comment (alinush) [inline crates/aptos-dkg/src/dlog/bsgs.rs:43]: Cute function name! ❤️

File: crates/aptos-dkg/src/dlog/bsgs.rs [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    let mut table = HashMap::with_capacity(m as usize);
+    let mut current = C::zero();
+
+    for j in 0..m {
+        let mut buf = [0u8; N];
+        current.serialize_compressed(&mut &mut buf[..]).unwrap();
+        table.insert(buf, j);
+        current += G;
```
Comment (alinush) [inline crates/aptos-dkg/src/dlog/bsgs.rs:51]: Nice, this is fast!

File: crates/aptos-dkg/src/dlog/bsgs.rs [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+///
+/// Returns a HashMap: `C::Affine.to_compressed() |---> exponent`
+#[allow(non_snake_case)]
+pub fn build_baby_table<C: CurveGroup, const N: usize>(G: C, m: u64) -> HashMap<[u8; N], u64> {
+    let mut table = HashMap::with_capacity(m as usize);
+    let mut current = C::zero();
+
+    for j in 0..m {
+        let mut buf = [0u8; N];
+        current.serialize_compressed(&mut &mut buf[..]).unwrap();
```
Comment (alinush) [inline crates/aptos-dkg/src/dlog/bsgs.rs:49]: Two `&mut &mut`?

File: crates/aptos-dkg/src/dlog/bsgs.rs [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    use ark_ec::PrimeGroup;
+    use rand::{thread_rng, Rng};
+
+    const COMPRESSED_SIZE: usize = 48;
+
+    #[allow(non_snake_case)]
+    #[test]
+    fn test_bsgs_bn254_random() {
+        let G = G1Projective::generator();
+        let m = 1 << 8;
```
Comment (alinush) [inline crates/aptos-dkg/src/dlog/bsgs.rs]: `2^8 = 256`, so pretty small which is good!

But you should test all values $0 G, 1 G, 2 G, \ldots, (m^2 - 1)G$ here, not just 10 random ones. We wanna make sure there are no edge cases.

If the test runs too slow, decrease `m`.

File: crates/aptos-dkg/src/pcs/univariate_hiding_kzg.rs [hunk 8]
```diff
@@ -19,29 +22,29 @@ use ark_ec::{
...
 #[derive(CanonicalSerialize, CanonicalDeserialize, Debug, PartialEq, Eq, Clone)]
 pub struct OpeningProof<E: Pairing> {
-    pi_1: Commitment<E>,
-    pi_2: E::G1,
+    pub(crate) pi_1: Commitment<E>,
+    pub(crate) pi_2: E::G1,
+}
+
+impl<E: Pairing> OpeningProof<E> {
+    /// Generates a random looking transcript (but not a valid one).
```
Comment (alinush) [inline crates/aptos-dkg/src/pcs/univariate_hiding_kzg.rs]: Transcript?

File: crates/aptos-dkg/src/pvss/chunky/hkzg_chunked_elgamal.rs [hunk 9]
```diff
@@ -52,52 +58,87 @@ type LiftedHkzg<'a, E> =
...
+//                                  │   TupleHomomorphism output       │
+//                                  │   (pair of HKZG image and        │
+//                                  │    Chunked ElGamal image)        │
+//                                  └──────────────────────────────────┘
+//
 //
 // In other words, the tuple homomorphism is roughly given as follows:
 //
 // ( rho, z_{i,j} , r_j ) │----> ( HKZG(rho, (0, z_{i,j}) ) , chunked_elgamal( z_{i,j} , r_j )
+//                             = ( \xi_1 * \rho  + \sum_i,j \ell_{i * B + j + 1}(\tau)_1 * z_{i,j} ) ,
```
Comment (alinush) [inline crates/aptos-dkg/src/pvss/chunky/hkzg_chunked_elgamal.rs]: Please use $a \cdot G$ syntax, for consistency (you often flip it as $G\cdot a$).

Please reformat this more clearly as:
```
//                             = ( 
//			                         \rho [\xi]_1 + \sum_i,j z_i,j [\ell_{i * B + j + 1}(\tau)]_1,
//                                  ( z_i,j G_1 + r_j ek_i )_i,j,
//                                  ( r_j H_1 )_j,
//			                   )
```

File: crates/aptos-dkg/src/pvss/chunky/hkzg_chunked_elgamal.rs [hunk 10]
```diff
@@ -52,52 +58,87 @@ type LiftedHkzg<'a, E> =
...
 //
 // TODO: note here that we had to put a zero before z_{i,j}, because that's what DeKARTv2 is doing. So maybe
 // it would make more sense to say this is a tuple homomorphism consisting of (lifts of) the
 // DeKARTv2::commitment_homomorphism together with the chunked_elgamal::homomorphism.
 pub type Homomorphism<'a, E> = TupleHomomorphism<LiftedHkzg<'a, E>, LiftedChunkedElgamal<'a, E>>;
 
+pub type Proof<'a, E> = sigma_protocol::Proof<E, Homomorphism<'a, E>>;
+
+impl<'a, E: Pairing> Proof<'a, E> {
+    /// Generates a random looking transcript (but not a valid one).
```
Comment (alinush) [inline crates/aptos-dkg/src/pvss/chunky/hkzg_chunked_elgamal.rs]: A bit confused by your use of "transcript" here. Let's consistently refer to this as a sigma protocol proof, not a transcript. We already have notions of "Fiat-Shamir transcripts" and "PVSS transcripts." We should avoid adding more.

File: crates/aptos-dkg/src/pvss/chunky/keys.rs [hunk 11]
```diff
@@ -89,3 +89,10 @@ impl<E: Pairing> DealtPubKey<E> {
 #[allow(dead_code)]
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct DealtPubKeyShare<E: Pairing>(pub(crate) DealtPubKey<E>);
+
+impl<E: Pairing> DealtPubKeyShare<E> {
+    #[allow(dead_code)]
+    pub fn new(dealt_pk: DealtPubKey<E>) -> Self {
```
Comment (alinush) [inline crates/aptos-dkg/src/pvss/chunky/keys.rs:95]: What's the purpose of this?

File: crates/aptos-dkg/src/pvss/das/fiat_shamir.rs
Comment (alinush) [inline crates/aptos-dkg/src/pvss/das/fiat_shamir.rs:1]: Music to my ears 🎵

File: crates/aptos-dkg/src/pvss/das/unweighted_protocol.rs [hunk 12]
```diff
@@ -173,10 +175,11 @@ impl traits::Transcript for Transcript {
             );
         }
 
+        let mut rng = thread_rng();
         // Derive challenges deterministically via Fiat-Shamir; easier to debug for distributed systems
```
Comment (alinush) [inline crates/aptos-dkg/src/pvss/das/unweighted_protocol.rs]: Replace this comment with:

```
// Deriving challenges by flipping coins: less complex to implement & less likely to get wrong. Creates bad RNG risks but we deem that acceptable.
```

File: crates/aptos-dkg/src/pvss/das/unweighted_protocol.rs [hunk 13]
```diff
@@ -173,10 +175,11 @@ impl traits::Transcript for Transcript {
...
         // TODO: benchmark this
-        let (f, extra) =
-            fiat_shamir::derive_challenge_scalars(self, sc, pp, spks, eks, auxs, &Self::dst(), 2);
+        let f = random_scalars(sc.n + 1 - sc.t, &mut rng);
+        let extra = random_scalars(2, &mut rng);
```
[the remark this answers is not in the document]
Comment (alinush) [reply crates/aptos-dkg/src/pvss/das/unweighted_protocol.rs:180]: This is fine: 
- `f` is used as the random polynomial in the SCRAPE LDT test
- `extra[0]` is used as a random challenge to batch verify the SoK sigma protocols (by taking powers of it)
- `extra[1]` is used as a random challenge to batch verify the encryption pairing check (also by taking powers of it)

File: crates/aptos-dkg/src/pvss/das/unweighted_protocol.rs
Comment (alinush) [inline crates/aptos-dkg/src/pvss/das/unweighted_protocol.rs]: Can you use the `LowDegreeTest::random` method here, which will pick `f` itself please?

File: crates/aptos-dkg/src/pvss/das/weighted_protocol.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (alinush) [reply crates/aptos-dkg/src/pvss/das/weighted_protocol.rs:203]: [Same note](https://github.com/aptos-labs/aptos-core/pull/18049/files#r2504640202).

File: crates/aptos-dkg/src/pvss/das/weighted_protocol.rs
Comment (alinush) [inline crates/aptos-dkg/src/pvss/das/weighted_protocol.rs]: Same: use `random` which picks `f` itself.

File: crates/aptos-dkg/src/pvss/weighted/weighted_config.rs [hunk 14]
```diff
@@ -37,16 +37,6 @@ pub struct WeightedConfig {
     min_weight: usize,
 }
 
-impl ThresholdConfig for WeightedConfig {
```
Comment (alinush) [inline crates/aptos-dkg/src/pvss/weighted/weighted_config.rs:40]: Nice!

File: crates/aptos-dkg/src/range_proofs/dekart_univariate_v2.rs [hunk 15]
```diff
@@ -10,33 +10,52 @@ use crate::{
...
     D: E::G1,
     a: E::ScalarField,
     a_h: E::ScalarField,
-    a_js: Vec<E::ScalarField>,
+    a_js: Vec<E::ScalarField>, // has length ell
     pi_gamma: univariate_hiding_kzg::OpeningProof<E>,
 }
 
+impl<E: Pairing> Proof<E> {
+    /// Generates a random looking transcript (but not a valid one).
```
Comment (alinush) [inline crates/aptos-dkg/src/range_proofs/dekart_univariate_v2.rs]: Please call it a proof! (The struct is named `Proof` for a reason!)

File: crates/aptos-dkg/src/range_proofs/dekart_univariate_v2.rs [hunk 16]
```diff
@@ -698,9 +717,10 @@ impl<E: Pairing> traits::BatchedRangeProof<E> for Proof<E> {
                 .map(|(&a_j, &mu_j)| a_j * mu_j)
                 .sum::<E::ScalarField>();
 
+        use sigma_protocol::homomorphism::TrivialShape as HkzgCommitment;
         univariate_hiding_kzg::CommitmentHomomorphism::verify(
             vk_hkzg.clone(),
-            univariate_hiding_kzg::Commitment(U),
+            HkzgCommitment(U), // Ugh univariate_hiding_kzg::Commitment(U) does not work because it's a tuple struct, see https://github.com/rust-lang/rust/issues/17422; So make it a
```
Comment (alinush) [inline crates/aptos-dkg/src/range_proofs/dekart_univariate_v2.rs]: "make it a..."?

File: crates/aptos-dkg/src/range_proofs/dekart_univariate_v2.rs [hunk 17]
```diff
@@ -846,9 +866,27 @@ mod fiat_shamir {
...
     use super::*;
-    use crate::sigma_protocol::homomorphism::fixed_base_msms;
+    use crate::sigma_protocol::{homomorphism::fixed_base_msms, traits::FirstProofItem};
+    use aptos_crypto::arkworks::random::UniformRand;
     use aptos_crypto_derive::SigmaProtocolWitness;
     pub use sigma_protocol::homomorphism::TrivialShape as CodomainShape;
+    pub type Proof<E> = sigma_protocol::Proof<E, Homomorphism<E>>;
+
+    impl<E: Pairing> Proof<E> {
+        /// Generates a random looking transcript (but not a valid one).
```
Comment (alinush) [inline crates/aptos-dkg/src/range_proofs/dekart_univariate_v2.rs]: Proof!

File: crates/aptos-dkg/src/sigma_protocol/homomorphism/fixed_base_msms.rs [hunk 18]
```diff
@@ -153,3 +157,25 @@ where
...
     }
 }
+
+impl<E: Pairing, H, LargerDomain> sigma_protocol::Trait<E>
+    for homomorphism::LiftHomomorphism<H, LargerDomain>
+where
+    H: sigma_protocol::Trait<E>,
+    LargerDomain: Witness<E>,
+{
+    fn dst(&self) -> Vec<u8> {
```
Comment (alinush) [inline crates/aptos-dkg/src/sigma_protocol/homomorphism/fixed_base_msms.rs:167]: I think we gotta be a bit careful here. It may be easier to use hashing to ensure non-ambiguity.

Not saying what you have is wrong, but it may be harder to reason about.

Comment (alinush) [reply crates/aptos-dkg/src/sigma_protocol/homomorphism/fixed_base_msms.rs:167]: In general, please also take a look at my thoughts [here](https://github.com/aptos-labs/aptos-core/pull/17660#discussion_r2401981701) regarding domain separation & let's discuss a little.

I think out of those 3 things we are missing a session ID which we may want to add.

File: crates/aptos-dkg/src/sigma_protocol/traits.rs [hunk 19]
```diff
@@ -44,12 +41,15 @@ pub trait Trait<E: Pairing>:
     }
 
     #[allow(non_snake_case)]
-    fn verify(
+    fn verify<H>(
         &self,
         public_statement: &Self::Codomain,
-        proof: &Proof<E, Self>,
+        proof: &Proof<E, H>, // Would like to set &Proof<E, Self>, but that ties the lifetime of H to that of Self, but we'd like it to be eg static
```
Comment (alinush) [inline crates/aptos-dkg/src/sigma_protocol/traits.rs:47]: How come this was not a problem so far?

Comment (alinush) [reply crates/aptos-dkg/src/sigma_protocol/traits.rs:47]: (Perpetually worried about how this abstraction is getting out of hand... 😨 )

File: crates/aptos-dkg/src/sigma_protocol/traits.rs [hunk 20]
```diff
@@ -232,6 +232,29 @@ where
...
 }
 
+impl<E: Pairing, H: homomorphism::Trait> Proof<E, H>
+where
+    H::Domain: Witness<E>,
+    H::Codomain: Statement,
+{
+    /// No-op: circumvents the fact that proofs inherit the homomorphism’s lifetime. This method does nothing at runtime.
+    #[allow(non_snake_case)]
+    pub fn change_lifetime<H2>(self) -> Proof<E, H2>
```
Comment (alinush) [inline crates/aptos-dkg/src/sigma_protocol/traits.rs:242]: I am so confused right now...

File: crates/aptos-dkg/src/fiat_shamir.rs [hunk 21]
```diff
@@ -18,7 +18,8 @@ use ark_serialize::CanonicalSerialize;
 use ff::PrimeField as FfPrimeField;
 use serde::Serialize;
 
-pub const PVSS_DOM_SEP: &[u8; 21] = b"APTOS_SCRAPE_PVSS_DST"; // TODO: Name needs work, but check backwards-compatibility
+#[allow(dead_code)] // Will be used in the new PVSS
+pub const PVSS_DOM_SEP: &[u8; 26] = b"APTOS_PVSS_FIAT_SHAMIR_DST";
```
Comment (alinush) [inline crates/aptos-dkg/src/fiat_shamir.rs:21]: Are you breaking anything?

File: crates/aptos-dkg/src/fiat_shamir.rs [hunk 22]
```diff
@@ -106,8 +108,10 @@ impl<S: FromBytes> ScalarProtocol<S> for merlin::Transcript {
     }
 }
 
+// TODO: it may make sense to make an associated type Scalar of Transcript, then remove S here and replace it with T::Scalar. Or remove it entirely given that we now only use arkworks...
```
Comment (alinush) [inline crates/aptos-dkg/src/fiat_shamir.rs]: If stuff is not going to be used. Remove it.

**Review by alinush — COMMENTED** [review]
Let's discuss!

File: crates/aptos-dkg/src/pvss/das/unweighted_protocol.rs (hunk 12 above)
Comment (alinush) [reply crates/aptos-dkg/src/pvss/das/unweighted_protocol.rs]: I think you need to do the same in `weighted_protocol.rs`

File: crates/aptos-crypto/src/arkworks/random.rs (hunk 2 above)
Comment (waamm) [reply crates/aptos-crypto/src/arkworks/random.rs:33]: In the `generate()` functions

File: crates/aptos-dkg/src/dlog/bsgs.rs (hunk 3 above)
Comment (waamm) [reply crates/aptos-dkg/src/dlog/bsgs.rs:15]: Since it's constant, I don't think so. So that makes it a bit clumsy and I've removed it from my current code, performance gains should be absolutely negligible anyway.

File: crates/aptos-dkg/src/sigma_protocol/traits.rs (hunk 19 above)
Comment (waamm) [reply crates/aptos-dkg/src/sigma_protocol/traits.rs:47]: > How come this was not a problem so far?

It was, this stuff is largely inspired by the old PR.

File: crates/aptos-dkg/src/fiat_shamir.rs (hunk 21 above)
Comment (waamm) [reply crates/aptos-dkg/src/fiat_shamir.rs:21]: It's dead code so I don't think so?

File: crates/aptos-dkg/src/sigma_protocol/traits.rs (hunk 20 above)
Comment (waamm) [reply crates/aptos-dkg/src/sigma_protocol/traits.rs:242]: Homomorphisms often carry references to public parameters, but the Proof does not. So the lifetime of the Proof should not depend on that, but for some reason Rust doesn't think that way and ties the lifetime of Proof to its generic parameters.

A different workaround would be to make Proof generic over eg <E, H::Domain, H::Codomain> instead, but that seems uglier to me.

File: crates/aptos-dkg/src/pvss/chunky/hkzg_chunked_elgamal.rs (hunk 9 above)
Comment (waamm) [reply crates/aptos-dkg/src/pvss/chunky/hkzg_chunked_elgamal.rs]: The reason I'm doing $G \cdot a$ is because that's the order the code wants... sure you want it the other way around?

File: crates/aptos-dkg/src/range_proofs/dekart_univariate_v2.rs (hunk 16 above)
Comment (waamm) [reply crates/aptos-dkg/src/range_proofs/dekart_univariate_v2.rs]: ... a struct with one named field, like `a`

File: crates/aptos-dkg/src/pvss/weighted/weighted_config.rs [hunk 23]
```diff
@@ -23,7 +23,7 @@ use std::fmt::{Display, Formatter};
 pub struct WeightedConfig {
     /// A weighted config is a $w$-out-of-$W$ threshold config, where $w$ is the minimum weight
     /// needed to reconstruct the secret and $W$ is the total weight.
-    tc: ThresholdConfigBlstrs,
+    pub(crate) tc: ThresholdConfigBlstrs,
```
Comment (rex1fernando) [inline crates/aptos-dkg/src/pvss/weighted/weighted_config.rs]: PR description says ThresholdConfig is removed from WeightedConfig, but it looks like it's still here? (And looks like it's a `blstrs`-specific struct?)

Comment (waamm) [reply crates/aptos-dkg/src/pvss/weighted/weighted_config.rs]: There's a trait called `ThresholdConfig` which Alin wanted to remove here because it came with a `new()` method that didn't make much sense in this context (so could've alternatively be made part of a subtrait); the field is still there.

File: crates/aptos-crypto/src/arkworks/random.rs (hunk 2 above)
Comment (alinush) [reply crates/aptos-crypto/src/arkworks/random.rs:33]: ...which generate dummy proofs! 👍

File: crates/aptos-dkg/src/dlog/bsgs.rs (hunk 3 above)
Comment (alinush) [reply crates/aptos-dkg/src/dlog/bsgs.rs:15]: Ah, this will be patched in the next PR then.

File: crates/aptos-dkg/src/pvss/chunky/hkzg_chunked_elgamal.rs (hunk 9 above)
Comment (alinush) [reply crates/aptos-dkg/src/pvss/chunky/hkzg_chunked_elgamal.rs]: Fair enough but let's stay consistent w/ literature.

File: crates/aptos-dkg/src/pvss/chunky/keys.rs (hunk 11 above)
Comment (alinush) [reply crates/aptos-dkg/src/pvss/chunky/keys.rs:95]: Maybe remove it if not needed.

File: crates/aptos-dkg/src/fiat_shamir.rs (hunk 21 above)
Comment (alinush) [reply crates/aptos-dkg/src/fiat_shamir.rs:21]: Remove then?

**Review by alinush — APPROVED** [review]

File: crates/aptos-dkg/src/dlog/bsgs.rs [hunk 24]
```diff
@@ -36,51 +32,47 @@ pub fn baby_step_giant_step<C: CurveGroup, const N: usize>(
     None
 }
 
-/// Build a baby-step table of size `m`
-///
-/// Returns a HashMap: `C::Affine.to_compressed() |---> exponent`
 #[allow(non_snake_case)]
-pub fn build_baby_table<C: CurveGroup, const N: usize>(G: C, m: u64) -> HashMap<[u8; N], u64> {
```
Comment (alinush) [inline crates/aptos-dkg/src/dlog/bsgs.rs]: oh noes, where did the baby table go? 😮

File: crates/aptos-dkg/src/dlog/bsgs.rs [hunk 25]
```diff
@@ -36,51 +32,47 @@ pub fn baby_step_giant_step<C: CurveGroup, const N: usize>(
...
+        let m = 1 << 4;
 
-        let mut rng = thread_rng();
+        let baby_table = dlog::table::build::<G1Projective>(G, m);
 
-        // Test 10 random values of x < m*m (to stay within reasonable range)
-        for _ in 0..10 {
-            let x: u64 = rng.gen_range(0, m * m);
-            let H = G * ark_bn254::Fr::from(x);
+        // Test all values of x from 0 to m-1
```
Comment (alinush) [inline crates/aptos-dkg/src/dlog/bsgs.rs]: Typo in comment: `m^2 - 1`
